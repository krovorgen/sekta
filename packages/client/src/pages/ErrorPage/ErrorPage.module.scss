@import "../../scss/helpers/variables";
@import "../../scss/helpers/mixins";

.root {
  display: flex;
  align-items: center;
  justify-content: center;
  background-image: url("../../assets/images/AuthLayout/bg.jpg");
  min-height: 100vh;
  background-size: cover;
  background-repeat: no-repeat;
}

.container {
  display: flex;
  flex-direction: column;
  align-items: center;
}
