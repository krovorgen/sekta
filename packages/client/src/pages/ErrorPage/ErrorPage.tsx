import { FC } from 'react'
import { useNavigate, useRouteError } from 'react-router-dom'
import { isRouteErrorResponse } from 'react-router'

import { Typography } from '@alfalab/core-components/typography/cssm'
import { Gap } from '@alfalab/core-components/gap/cssm'
import { Button } from '@alfalab/core-components/button'

import st from './ErrorPage.module.scss'

type ErrorResponse = {
  status: number
  message?: string
}

export const ErrorPage: FC = () => {
  const navigate = useNavigate()
  const error = useRouteError() as ErrorResponse

  let errorMessage = 'Something went wrong'

  if (isRouteErrorResponse(error)) {
    switch (error.status) {
      case 401:
        errorMessage = "You aren't authorized to see this"
        break
      case 404:
        errorMessage = 'Page not found'
        break
      case 418:
        errorMessage = 'ü´ñ'
        break
      case 500:
        errorMessage = 'Internal Server Error'
        break
      case 503:
        errorMessage = 'Looks like our API is down'
        break
      default:
        break
    }
  } else if (error instanceof Error) {
    errorMessage = error.message
  }

  return (
    <div className={st.root}>
      <div className={st.container}>
        <Typography.Title tag="h1" color="static-secondary-dark" view="xlarge">
          {error?.status}
        </Typography.Title>
        <Gap size="xs" />
        <Typography.Title tag="h2" color="static-secondary-dark" view="small">
          {errorMessage}
        </Typography.Title>
        <Gap size="xs" />
        <Button onClick={() => navigate(-1)} view="link" size="xxs">
          –í–µ—Ä–Ω—É—Ç—å—Å—è
        </Button>
      </div>
    </div>
  )
}
