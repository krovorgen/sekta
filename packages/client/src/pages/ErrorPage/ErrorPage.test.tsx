import React from 'react'
import '@testing-library/jest-dom'
import { render, screen, fireEvent } from '@testing-library/react'
import {
  MemoryRouter,
  isRouteErrorResponse,
  useNavigate,
  useRouteError,
} from 'react-router-dom'
import { ErrorPage } from './ErrorPage'

// Mock the useNavigate hook
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn(),
}))

// Mock the useRouteError hook
jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useRouteError: jest.fn(),
  isRouteErrorResponse: jest.fn(),
}))

describe('ErrorPage component', () => {
  beforeEach(() => {
    // Reset mocks and provide a default error value
    ;(useNavigate as jest.Mock).mockClear()
    ;(useRouteError as jest.Mock).mockClear()
    ;(useRouteError as jest.Mock).mockReturnValue({
      status: 404,
      message: 'Page not found',
    })
    ;(isRouteErrorResponse as unknown as jest.Mock).mockReturnValue(true)
  })

  test.skip('Snapshot', () => {
    const wrapper = render(
      <MemoryRouter>
        <ErrorPage />
      </MemoryRouter>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the correct error message', () => {
    render(
      <MemoryRouter>
        <ErrorPage />
      </MemoryRouter>
    )

    const errorMessage = screen.getByText('Page not found')
    expect(errorMessage).toBeInTheDocument()
  })

  it('renders a "Вернуться" button', () => {
    render(
      <MemoryRouter>
        <ErrorPage />
      </MemoryRouter>
    )

    const backButton = screen.getByText('Вернуться')
    expect(backButton).toBeInTheDocument()
  })

  it('calls the navigate function when the "Вернуться" button is clicked', () => {
    const mockNavigate = jest.fn()
    ;(useNavigate as jest.Mock).mockReturnValue(mockNavigate)

    render(
      <MemoryRouter>
        <ErrorPage />
      </MemoryRouter>
    )

    const backButton = screen.getByText('Вернуться')
    fireEvent.click(backButton)

    expect(mockNavigate).toHaveBeenCalledWith(-1)
  })
})
