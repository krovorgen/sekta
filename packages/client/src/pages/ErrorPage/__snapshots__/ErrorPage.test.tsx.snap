// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage component Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <h1
            class=""
          >
            404
          </h1>
          <div
            class=""
            data-gap-size="xs"
          />
          <h2
            class=""
          >
            Page not found
          </h2>
          <div
            class=""
            data-gap-size="xs"
          />
          <button
            class="button__component_5v2qr button__link_5v2qr button__xxs_5v2qr button__xxs_1zzy9 button__component_18dw8 button__link_18dw8 button__component_1x5x4 button__link_1x5x4"
            type="button"
          >
            <span
              class="button__text_5v2qr button__stretchText_5v2qr"
            >
              Вернуться
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <h1
          class=""
        >
          404
        </h1>
        <div
          class=""
          data-gap-size="xs"
        />
        <h2
          class=""
        >
          Page not found
        </h2>
        <div
          class=""
          data-gap-size="xs"
        />
        <button
          class="button__component_5v2qr button__link_5v2qr button__xxs_5v2qr button__xxs_1zzy9 button__component_18dw8 button__link_18dw8 button__component_1x5x4 button__link_1x5x4"
          type="button"
        >
          <span
            class="button__text_5v2qr button__stretchText_5v2qr"
          >
            Вернуться
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
